src/devtools/protocol.ts

```diff
@@ -87,25 +76,15 @@ export type SDKToPanelMessage =
   | {
       msg: 'doc::sync::full';
       docKey: string;
-      root: Devtools.JSONElement;
-      clients: Array<Devtools.Client>;
+      events: Array<Array<DocEvent>>;
     }
   /**
-   * Sent whenever the document is updated.
+   * Sent whenever the document is changed.
    */
   | {
       msg: 'doc::sync::partial';
       docKey: string;
-      event?: DocEvent;
-      root?: Devtools.JSONElement;
-      clients?: Array<Devtools.Client>;
-    }
-  /**
-   * Sent detailed information for the node corresponding to the given path.
-   */
-  | {
-      msg: 'doc::node::detail';
-      node: Devtools.TreeNodeInfo;
+      event: Array<DocEvent>;
```

Comment:
It seems that `Array<DocEvent>` represents a `Change`, and each `DocEvent` represents an `Operation`. I am concerned about the low readability. Is there any good way?

```ts
events: Array<Array<DocEvent>>;
event: Array<DocEvent>;
```

Reply:
I agree with your concern, and I think defining a separate type for Array<DocEvent> would be good. While the name might require some consideration, for now, I've modified it to TransactionDocEvents. Please review and share your thoughts.